common/app/common/ApplicationMode.scala

```diff
@@ -0,0 +1,19 @@
+package common
+
+import play.api.{Application, GlobalSettings}
+import play.api.Mode._
+
+object ApplicationMode {
+  private var theMode = Prod
```

Comment:
I don't think Option works, here. Surely we need to simplify things, so that any code that references the mode, now has access to the application?